server/dash/dashboards/nc_summary_comparison.py

```diff
@@ -0,0 +1,754 @@
+            ("Quality Filter: On" or "Quality Filter: Off").
+    """
+    df = generate_filtered_dataframe(api_data_df, nc_dropdown, nc_dropdown_filter)
+    df, num_bins, data_quality_output = filter_bad_quality_data(df, data_quality_switch)
+
+    # Distribution for the total number of requests.
+    print(" * Generating requests time to close histogram")
+    time_close_histogram = px.histogram(df, x="timeToClose",
+    title="Distribution of Time to Close Request",
+     nbins=num_bins, range_x=[min(
+         df.loc[:, "timeToClose"]), max(df.loc[:, "timeToClose"])],
+        labels={"timeToClose": "Request Duration",
+         "count": "Frequency"})
+    time_close_histogram.update_layout(margin=dict(l=50, r=50, b=50, t=50), font=dict(size=9))
+    return time_close_histogram, data_quality_output
+
+
+@callback(
+    Output("num_req_line_chart", "figure"),
+    Input("nc_dropdown", "value"),
+    [Input("nc_dropdown_filter", "value")],
+    Input("data_quality_switch", "value")
+)
+def generate_overlap_req_line_charts(nc_dropdown, nc_dropdown_filter=None,
+data_quality_switch=True):
+    """Generates the summary visualizations for LA 311 requests data based 
+    on selected neighborhood conucil, request types, and data quality switch.
+
+    This function takes the selected neighborhood council (nc) value from the
+     "nc_dropdown" dropdown, selected request type from "nc_dropdown_filter"
+    dropdown, and the status on the "data_quality_switch" toggle to output a 
+    line chart showing request overtime.
+
+    Args:
+        nc_dropdown: A string argument automatically detected by Dash callback
+            function when "nc_dropdown" element is selected in the layout.
+        nc_dropdown_filter: A list of strings automatically detected by Dash 
+            callback function when "nc_dropdown_filter" element is selected in the 
+            layout, default None.
+        data_quality_switch: A boolean for data quality filter automatically 
+            detected by Dash callback function when "data_quality_switch" element
+             is selected in the layout, default True.
+    Returns: 
+        Line chart showing the number of requests throughout the day.
+    """
+    df = generate_filtered_dataframe(api_data_df, nc_dropdown, nc_dropdown_filter)
+    df, _, _ = filter_bad_quality_data(df, data_quality_switch)
+
+    # Time Series for the Total Number of Requests.
+    print(" * Generating number of requests line chart")
+    req_time = pd.DataFrame(df.groupby("createDateDT", as_index=False)["srnumber"].count())
+    num_req_line_chart = px.line(req_time, x="createDateDT", y="srnumber",
+    title="Total Number of 311 Requests Overtime", labels={
+        "createDateDT": "DateTime", "srnumber": "Frequency"})
+    num_req_line_chart.update_layout(margin=dict(l=25, r=25, b=25, t=50), font=dict(size=9))
+    return num_req_line_chart
+
+
+def generate_two_filtered_df(api_data_df, nc_comp_dropdown, nc_comp_dropdown2):
```

Comment:
I think it makes more sense to make this `generate_filtered_df` and then call it twice, one for each selected NC.

Reply:
Thanks for the suggestion - resolved